src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -273,28 +312,75 @@ def get(self, request: Request, project: Project) -> Response:
                     )
                     and project.organization.flags.codecov_access
                 )
+                codecov_enabled = True
                 # Check the cache and skip querying Codecov if a query within the last hour found no integration.
                 codecov_cache = cache.get(cache_key.format(project_id=project.id))
                 if codecov_enabled and codecov_cache is False:
                     result["codecov"] = {"status": 404}
 
                 elif codecov_enabled and codecov_cache is not False:
+                    # Get Codecov data
                     try:
-                        lineCoverage, codecovUrl = get_codecov_data(
-                            repo=current_config["config"]["repoName"],
-                            service=current_config["config"]["provider"]["key"],
-                            branch=current_config["config"]["defaultBranch"],
-                            path=current_config["outcome"]["sourcePath"],
+                        ref = (
+                            ctx["commit_id"]
+                            if ctx["commit_id"]
+                            else current_config["config"]["defaultBranch"]
```

Comment:
Should we be getting rid of this line? (since we're not going to use the branch as the fallback)

Reply:
Right now, getting the latest commit and using it is behind a flag, as Snighda suggested. This way, we can test it out before exposing it to customers.

[Here](https://github.com/getsentry/sentry/pull/43850/files#diff-9ed2a929b59278cbb7d160ac361180e90dbb3ec7359f7543b614e12e1615d059R363-R366) is where we prevent line coverage from being shown for orgs where the `codecov_commit_sha_from_git_blame_enabled` flag is enabled and the commit sha was not successfully found from git